doc/docs/modules/ROOT/pages/quickstart_writer.adoc

```diff
@@ -215,4 +215,108 @@ UNwIND $events AS event
 CREATE (n:Person{fullName: event.name + event.surname})
 ----
 
-Where `event` represents each Dataset row.
\ No newline at end of file
+Where `event` represents each Dataset row.
+
+=== Write data by Relationship Type
+
+You can write a dataframe to Neo4j by specifying source, target and relation.
+There are two strategies you can use to write relationships: *NATIVE* and *KEYS*
+
+==== Native Strategy
+
+This strategy is useful when you have a schema that conforms with the <<quickstart_reader.adoc#bookmark-rel-read-schema,Relationship Read Schema>>
+
+```scala
+df.write
+    .format("org.neo4j.spark.DataSource")
+    .option("url", "bolt://localhost:7687")
+    .option("relationship", "SOLD")
+    .option("relationship.source.labels", ":Person")
+    .option("relationship.source.save.mode", "Overwrite")
+    .option("relationship.target.labels", ":Product")
+    .option("relationship.target.save.mode", "Overwrite")
+    .save()
+```
+
+You just need to specify the source node labels, the target node labels, and the relationship you want between them.
+
+[NOTE]
+The default save mode for source and target nodes is `match`. This means that the relationship will be created only if the nodes are already in your DB. Look at <<node-save-modes,here>> for more info about node save modes.
+
+Using `Overwrite` or `Match` node save mode, you can speficy which keys should be used to write the nodes.
```

Comment:
`Overwrite` is more of a Spark terminology as you can see [here](https://spark.apache.org/docs/2.3.0/sql-programming-guide.html#save-modes). We just extended it to add the `Match` mode since it seemed more consistent with Spark. If you think it can cause confusion we can do one of:

- improve the docs to explain why is `Overwrite` and not `Merge` and what does it do
- change the Node Save Modes to neo4j terminology

@conker84 do you have other suggestions?

Reply:
@moxious, as @utnaf said, we tried to stick to Spark's terminology, because we manage the `mode`.
Node scenario 1:
```scala
ds.write
      .format(classOf[DataSource].getName)
      .mode(SaveMode.ErrorIfExists) // mode
      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
      .option("labels", ":MyNode:MyLabel")
      .save()
```
we create the following query:

```cypher
UNWIND $batch AS event CREATE (n:MyNode:MyLabel) SET n += event.properties
```

Node scenario 1:
```scala
ds.write
      .format(classOf[DataSource].getName)
      .mode(SaveMode.Overwrite) // mode
      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
      .option("labels", ":Person:Customer")
      .option("node.keys", "surname")
      .save()
```
we create the following query:

```cypher
UNWIND $batch AS event MERGE (n:Person:Customer{surname: event.keys.surname}) SET n += event.properties
```

So we thought that adding the same syntax for relationships would be the best solution.
Relationship scenario 1:
```scala
dfOriginal.write
      .mode(SaveMode.Append) // mode
      .format(classOf[DataSource].getName)
      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
      .option("relationship", "SOLD")
      .option("relationship.save.strategy", "NATIVE")
      .option("relationship.source.labels", ":Person")
      .option("relationship.target.labels", ":Product")
      .option("relationship.source.node.keys", "source.id:id")
      .option("relationship.target.node.keys", "target.id:id")
      .option("relationship.source.save.mode", "Overwrite")
      .option("relationship.target.save.mode", "Match")
      .save()
```

we create the following query:

```cypher
UNWIND $batch AS event 
MERGE (source:Person{id: event.source.keys.id}) SET n += event.source.properties
MATCH (target:Product{surname: event.target.keys.id})
CREATE (source)-[rel:SOLD]->(target) SET rel += event.properties
```

Relationship scenario 1:
```scala
dfOriginal.write
      .mode(SaveMode.Overwrite) // mode
      .format(classOf[DataSource].getName)
      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
      .option("relationship", "SOLD")
      .option("relationship.save.strategy", "NATIVE")
      .option("relationship.source.labels", ":Person")
      .option("relationship.target.labels", ":Product")
      .option("relationship.source.node.keys", "source.id:id")
      .option("relationship.target.node.keys", "target.id:id")
      .option("relationship.source.save.mode", "Overwrite")
      .option("relationship.target.save.mode", "Match")
      .save()
```

we create the following query:

```cypher
UNWIND $batch AS event 
MERGE (source:Person{id: event.source.keys.id}) SET n += event.source.properties
MATCH (target:Product{surname: event.target.keys.id})
MERGE (source)-[rel:SOLD]->(target) SET rel += event.properties
```